src/compiler/checker.ts

```diff
@@ -7195,12 +7196,15 @@ namespace ts {
                             : Diagnostics.Generic_type_0_requires_between_1_and_2_type_arguments;
                     const typeStr = typeToString(type, /*enclosingDeclaration*/ undefined, TypeFormatFlags.WriteArrayAsGenericType);
                     error(node, diag, typeStr, minTypeArgumentCount, typeParameters.length);
-                    return unknownType;
+                    if (!isJs) {

```

Comment:
so what broke with the permissive behavior in TS?

Reply:
Nothing is "broken" once all the arity-based signature removals are excised and arity verification is moved into `checkTypeArguments` - just observable behavior changes to follow-on error scenarios (which cause a bunch of baseline churn - and in TS it is not _uniformly_ better behavior, since we fill with `{}` instead of `any` - we could change that if there's an arity mismatch); I'll open a PR on my PR to show you, here: https://github.com/weswigham/TypeScript/pull/2

I can merge it into this PR and iterate on it if you think it's a good idea (or do a second PR after this one). I just didn't want to add it in (it is more complicated, but also likely more complete) without mentioning it to someone first.

...That took longer than I expected to fully flesh out - partially because I had to work around our inconsistent handling of generics on psuedoconstructors (like `ArrayConstructor`).

Reply:
thanks for the clarification. i think i understand now. i am not sure i understand the change in https://github.com/weswigham/TypeScript/pull/2, but we should talk about that one on Monday.